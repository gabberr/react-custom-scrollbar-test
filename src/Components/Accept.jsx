import React, { Component } from 'react';
import Scrollbar from 'react-scrollbars-custom'; //Scrollbar fra https://www.npmjs.com/package/react-scrollbars-custom & https://xobotyi.github.io/react-scrollbars-custom/


export default class Accept extends Component {
    render() {
        return (
            <div>
                <h1 style={{position: 'static',  left: '50%'}}>The light has been </h1>
                <Scrollbar style={{ width: 200, height: 40, alignContent:'center', position: 'absolute', left: '45%'}}>
                    <div style={{ marginLeft: '0px', marginRight: '0px' }}>
                        <div>
                            {this.props.states.greenText}
                        </div>
                        <div>
                            {this.props.states.yellowText}
                        </div>
                        <div>
                            {this.props.states.redText}
                        </div>
                    </div>
                    </Scrollbar>
                <div></div>
            </div>
        )
    }
}