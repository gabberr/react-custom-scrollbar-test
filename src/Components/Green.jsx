import React, { Component } from 'react';   

export default class Green extends Component{
render() {              
  return (    
    <div>
      {this.props.green}
    </div>
  )
}  
}