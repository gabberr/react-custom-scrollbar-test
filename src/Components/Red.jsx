import React, { Component } from 'react'; 

export default class Red extends Component {   
   render() {       
   return (
     <div>
     {this.props.red}
     </div>
   )
   }
  }