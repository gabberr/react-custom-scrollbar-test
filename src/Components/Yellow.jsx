import React, { Component } from 'react'; 

export default class Yellow extends Component {      
  render() {          
    return (  
      <div> 
      {this.props.yellow} 
      </div>
    )   
  } 
}