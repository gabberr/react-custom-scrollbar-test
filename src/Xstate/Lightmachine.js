export const lightMachine = {
  id: 'light',
  initial: 'step1',
  states: {
    step1: {
      on: {
        NEXT: 'step2'
      }
    },
    step2: {
      on: {
        NEXT: 'step3'
      }
    },
    step3: {
      on: {
        NEXT: 'step4'
      }
    },
    step4:{
      on:{
        NEXT:'step4'
      }
    }
  }
};