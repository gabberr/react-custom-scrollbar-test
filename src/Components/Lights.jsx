import React, { Component } from "react"; //import of React
import { lightMachine } from "../Xstate/Lightmachine"; //import statemachine
import { State, withStateMachine } from "react-automata";
//#region import of components
import  Green  from "./Green";
import Yellow from './Yellow';
import Red from './Red';
import Accept from "./Accept";
//#endregion

export class Lights extends Component {
  render() {
    return (
      <div>

        <State is="step1">
 
          <Green green={this.props.states.greenText}></Green>
        </State>

        <State is="step2">
          yellow
          <Yellow yellow={this.props.states.yellowText}></Yellow>
        </State>

        <State is="step3" >
          red
          <Red red={this.props.states.redText}></Red>
        </State>
        <State is="step4">
          <Accept states={this.props.states}></Accept>
        </State>
        
        <button
        style={{position: 'absolute', top:'140px',left: '48%'}}
            className="btn-back"
            type="submit"
            onClick={() => {
              this.transitionToState("NEXT");
            }}
          >
          click here
        </button>   
      </div>
    );
  }

  transitionToState = (nextTransition, nextViewstate = {}) => {
    console.log(`Existing State ${this.props.machineState.value}`);
    var statesAvailable = lightMachine.states[this.props.machineState.value].on;
    console.log(`transitionToState available ${JSON.stringify(statesAvailable)} 
              nextTransition ${nextTransition}`);
    if (Object.keys(statesAvailable).includes(nextTransition)) {
      if (!this.inTransition) {
        console.log(`Transition to: ${JSON.stringify(nextTransition)}`);
        var nextState = statesAvailable[nextTransition];
        try {
          console.log(
            `Transition: ${nextTransition} NextState: ${nextState.value}`
          );
          this.props.transition(nextTransition);
        } catch (err) {
          console.log("Transition failed");
        }
      } else {
        console.log(
          `Cannot transition to ${nextTransition} as we are in active transition. We should wait a tick`
        );
      }
    } else {
      console.log(
        `State ${
          this.props.machineState.value
        } cannot transition to ${nextTransition}`
      );
    }
  };
}
export default withStateMachine(lightMachine)(Lights);
