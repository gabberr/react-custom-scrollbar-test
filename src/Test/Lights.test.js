
/*
test("lights with imported components", () => {
  testStateMachine({ lightMachine }, Lights);
});
*/

import { testStateMachine, withStateMachine } from 'react-automata';
import { Lights } from '../Components/Lights';
import { lightMachine } from "../Xstate/Lightmachine";

test('lights with imported components', () => {

  const fixtures = {
    initialData: {
      states: {
        'greenText': 'green light',
        'yellowText':'yellow light',
        'redText':'red light'
      }
    }
  }

  const StateMachine = withStateMachine(lightMachine)(Lights)

  testStateMachine(StateMachine, { fixtures });
});