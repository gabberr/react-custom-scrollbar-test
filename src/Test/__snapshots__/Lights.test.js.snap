// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lights with imported components: step1 1`] = `
<div>
  <div>
    green light
  </div>
  <button
    className="btn-back"
    onClick={[Function]}
    style={
      Object {
        "left": "48%",
        "position": "absolute",
        "top": "140px",
      }
    }
    type="submit"
  >
    click here
  </button>
</div>
`;

exports[`lights with imported components: step2 1`] = `
<div>
  yellow
  <div>
    yellow light
  </div>
  <button
    className="btn-back"
    onClick={[Function]}
    style={
      Object {
        "left": "48%",
        "position": "absolute",
        "top": "140px",
      }
    }
    type="submit"
  >
    click here
  </button>
</div>
`;

exports[`lights with imported components: step3 1`] = `
<div>
  red
  <div>
    red light
  </div>
  <button
    className="btn-back"
    onClick={[Function]}
    style={
      Object {
        "left": "48%",
        "position": "absolute",
        "top": "140px",
      }
    }
    type="submit"
  >
    click here
  </button>
</div>
`;
